### PR TITLE
Improve onboarding experience by handling common failure scenarios for new users more gracefully

### PR DESCRIPTION
**Pull request by practicalswift**
Improve onboarding experience by handling common failure scenarios for new users more gracefully.

Improve usability in these scenarios:
* `bitcoin-cli` not available in `PATH` and/or `bitcoind` not installed
* `bitcoin-cli` available in `PATH` but `bitcoind` is not running

Before this commit (scenario: `bitcoin-cli` not available in PATH and/or `bitcoind` not installed):

```
$ lightningd/lightningd
bitcoin-cli exec failed: No such file or directory
Aborted (core dumped)
```

After this commit (scenario: `bitcoin-cli` not available in PATH and/or `bitcoind` not installed):

```
$ lightningd/lightningd
bitcoin-cli not found. Is bitcoin-cli (part of Bitcoin Core) available in your PATH?

Make sure you have bitcoind running and that bitcoin-cli is able to connect to bitcoind.

You can verify that your Bitcoin Core installation is ready for use by running:

    $ bitcoin-cli -testnet echo 'hello world'
```

Before this commit (scenario: `bitcoin-cli` available in `PATH` but `bitcoind` is not running):

```
$ lightningd/lightningd
bitcoin-cli exited with code 1: error: Could not locate RPC credentials. No authentication cookie could be found, and RPC password is not set.  See -rpcpassword and -stdinrpcpass.  Configuration file: (/root/.bitcoin/bitcoin.conf)

Aborted (core dumped)
```

After this commit (scenario: `bitcoin-cli` available in `PATH` but `bitcoind` is not running):

```
$ lightningd/lightningd
Could not connect to bitcoind using bitcoin-cli. Is bitcoind running?

Make sure you have bitcoind running and that bitcoin-cli is able to connect to bitcoind.

You can verify that your Bitcoin Core installation is ready for use by running:

    $ bitcoin-cli -testnet echo 'hello world'
```

### CODE REVIEW

File: lightningd/bitcoind.h [hunk 1]
```diff
@@ -61,6 +61,8 @@ struct bitcoind *new_bitcoind(const tal_t *ctx,
 			      struct lightningd *ld,
 			      struct log *log);
 
+void report_bitcoind_unavailability_and_exit(struct bitcoind *bitcoind, const char *error_message);
```
Comment (cdecker) [inline lightningd/bitcoind.h]: Why is this being exported? The function is afaik only being used from `bitcoin.c`.

File: lightningd/bitcoind.c [hunk 2]
```diff
@@ -716,6 +716,23 @@ static const char **cmdarr(const tal_t *ctx, const struct bitcoind *bitcoind,
 	return args;
 }
 
+void report_bitcoind_unavailability_and_exit(struct bitcoind *bitcoind, const char *error_message)
```
Comment (cdecker) [inline lightningd/bitcoind.c]: Quite an unwieldy name :wink:

**Comment by cdecker** [comment]
Excellent idea, we get way too many users reporting issues due to these scenarios, so providing help is big win here.

File: lightningd/bitcoind.c (hunk 2 above)
Comment (practicalswift) [reply lightningd/bitcoind.c]: Agreed! Any suggestions on naming? :-)

Comment (cdecker) [reply lightningd/bitcoind.c]: `fatal_bitcoind_failure`? It'd match our existing `fatal(msg)` functions.

**Comment by practicalswift** [comment]
@cdecker Updated! Please re-review :-)

**Comment by cdecker** [comment]
ACK ed80c46

**Comment by practicalswift** [comment]
Somewhat related – I have submitted [this PR](https://github.com/bitcoin/bitcoin/pull/12652) to Bitcoin Core in order to make the combined `c-lightning`/`bitcoind` onboarding experience less confusing :-)

**Pull request merged**